src/usr/share/gpgmailer/gpgkeyverifier.py

```diff
@@ -198,17 +186,18 @@ def _calculate_recipient_info(self, loop_current_time):
                 elif expiration_data['expiring_soon']:
                     expiring_soon_messages.append(expiration_data['warning_message'])
 
-                    valid_key_fingerprints.append(self.email_dicts[recipient_email]['fingerprint'])
-                    valid_recipient_emails.append(recipient_email)
+                    valid_key_fingerprints.append(fingerprint)
+                    for email in self.recipient_keys[fingerprint]['email_list']:
+                        valid_recipient_emails.append(email)
 
                 else:
-                    valid_key_fingerprints.append(self.email_dicts[recipient_email]['fingerprint'])
-                    valid_recipient_emails.append(recipient_email)
+                    valid_key_fingerprints.append(fingerprint)
+                    for email in self.recipient_keys[fingerprint]['email_list']:
+                        valid_recipient_emails.append(email)
 
-        # The sender key might be in valid_key_fingerprints despite not being a recipient. Check
-        #   valid_recipient_emails instead.
-        if valid_recipient_emails == []:
-            raise NoUsableKeysException('All recipient GPG keys have expired.')
+        # There should be at least one valid, non-expired fingerprint at this point.
+        if not valid_key_fingerprints:
+            raise NoUsableKeysException('All GPG keys have expired.')
```

Comment:
I'd probably have this read, "All possible recipient GPG keys have expired." That makes more sense to me and that is what is important. ;)